File: extract_schemes.js
const fs = require('fs');

const content = fs.readFileSync('cardsData.js', 'utf-8');

// Parse each card block by splitting on object boundaries
// Extract id, latest, schemeName per entry
const entries = [];
const idRegex = /"id"\s*:\s*(\d+)/g;
const blocks = content.split(/(?=\{\s*"id"\s*:)/);

for (const block of blocks) {
  const idMatch = block.match(/"id"\s*:\s*(\d+)/);
  const latestMatch = block.match(/"latest"\s*:\s*"([^"]+)"/);
  const schemeMatch = block.match(/"schemeName"\s*:\s*"([^"]+)"/);

  if (idMatch && schemeMatch) {
    entries.push({
      id: parseInt(idMatch[1]),
      schemeName: schemeMatch[1],
      latest: latestMatch ? latestMatch[1] : ''
    });
  }
}

// Unique by schemeName, preserving first occurrence order
const seen = new Set();
const unique = [];
for (const e of entries) {
  if (!seen.has(e.schemeName)) {
    seen.add(e.schemeName);
    unique.push(e);
  }
}

// Build CSV
const rows = [['Sr. No.', 'Scheme Name', 'Latest']];
unique.forEach((item, i) => rows.push([i + 1, item.schemeName, item.latest]));

const csv = rows.map(r => r.map(c => `"${String(c).replace(/"/g, '""')}"`).join(',')).join('\r\n');
fs.writeFileSync('unique_schemes.csv', '\uFEFF' + csv, 'utf-8'); // BOM for Excel

console.log(`Done! ${unique.length} unique schemes saved to unique_schemes.csv`);

File: extract_schemes.py
import re
import openpyxl

with open("cardsData.js", "r", encoding="utf-8") as f:
    content = f.read()

# Extract all schemeName values
schemes = re.findall(r'"schemeName"\s*:\s*"([^"]+)"', content)

# Unique, preserving order
seen = set()
unique_schemes = []
for s in schemes:
    if s not in seen:
        seen.add(s)
        unique_schemes.append(s)

# Write to Excel
wb = openpyxl.Workbook()
ws = wb.active
ws.title = "Scheme Names"

ws["A1"] = "Sr. No."
ws["B1"] = "Scheme Name"

# Style header
for cell in ws["1:1"]:
    cell.font = openpyxl.styles.Font(bold=True)

for i, name in enumerate(unique_schemes, start=1):
    ws.cell(row=i+1, column=1, value=i)
    ws.cell(row=i+1, column=2, value=name)

ws.column_dimensions["A"].width = 10
ws.column_dimensions["B"].width = 40

wb.save("unique_schemes.xlsx")
print(f"Done! {len(unique_schemes)} unique schemes saved to unique_schemes.xlsx")
